CMakeLists.txt

```diff
@@ -435,114 +484,91 @@ function(is_sdk_requested name result_var_name)
-  # FIXME: This only matches ARMv7l (by far the most common variant).
-  elseif("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "armv7l")
-    configure_sdk_unix(LINUX "Linux" "linux" "linux" "armv7" "armv7-unknown-linux-gnueabihf")
-    set(SWIFT_HOST_VARIANT_ARCH "armv7")
+    # FIXME: Is this correct? Should Android be the primary variant when it's a cross-compile target?
+    set(SWIFT_PRIMARY_VARIANT_SDK_default "ANDROID")
     set(SWIFT_PRIMARY_VARIANT_ARCH_default "armv7")
-  elseif("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "aarch64")
-    configure_sdk_unix(LINUX "Linux" "linux" "linux" "aarch64" "aarch64-unknown-linux-gnu")
-    set(SWIFT_HOST_VARIANT_ARCH "aarch64")
-    set(SWIFT_PRIMARY_VARIANT_ARCH_default "aarch64")
-  elseif("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "ppc64")
-    configure_sdk_unix(LINUX "Linux" "linux" "linux" "powerpc64" "powerpc64-unknown-linux-gnu")
-    set(SWIFT_HOST_VARIANT_ARCH "powerpc64")
-    set(SWIFT_PRIMARY_VARIANT_ARCH_default "powerpc64")
-  elseif("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "ppc64le")
-    configure_sdk_unix(LINUX "Linux" "linux" "linux" "powerpc64le" "powerpc64le-unknown-linux-gnu")
-    set(SWIFT_HOST_VARIANT_ARCH "powerpc64le")
-    set(SWIFT_PRIMARY_VARIANT_ARCH_default "powerpc64le")
-  else()
-    message(FATAL_ERROR "Unknown or unsupported architecture: ${CMAKE_SYSTEM_PROCESSOR}")
   endif()
 
-elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "FreeBSD")
+elseif("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "FREEBSD")
+  
+  set(CMAKE_EXECUTABLE_FORMAT "ELF")
+  set(SWIFT_HOST_VARIANT "freebsd" CACHE STRING
+      "Deployment OS for Swift host tools (the compiler) [freebsd].")
+
   # FIXME: Using the host OS version won't produce correct results for
   # cross-compilation.
   string(REPLACE "[-].*" "" FREEBSD_SYSTEM_VERSION ${CMAKE_SYSTEM_VERSION})
   message(STATUS "FreeBSD Version: ${FREEBSD_SYSTEM_VERSION}")
-  configure_sdk_unix(FREEBSD "FreeBSD" "freebsd" "freebsd" "x86_64"
+  configure_sdk_unix(FREEBSD "FreeBSD" "freebsd" "${SWIFT_HOST_VARIANT}" "x86_64"
     "x86_64-unknown-freebsd${FREEBSD_SYSTEM_VERSION}")
-
-  set(CMAKE_EXECUTABLE_FORMAT "ELF")
-
-  set(SWIFT_HOST_VARIANT "freebsd" CACHE STRING
-      "Deployment OS for Swift host tools (the compiler) [freebsd].")
-
-  set(SWIFT_HOST_VARIANT_SDK "FREEBSD")
-  set(SWIFT_HOST_VARIANT_ARCH "x86_64")
-
-  set(SWIFT_PRIMARY_VARIANT_SDK_default "FREEBSD")
-  set(SWIFT_PRIMARY_VARIANT_ARCH_default "x86_64")
-elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "CYGWIN")
-  configure_sdk_unix(CYGWIN "Cygwin" "windows" "cygwin" "x86_64" "x86_64-unknown-windows-cygnus")
-
-#  set(CMAKE_EXECUTABLE_FORMAT "ELF")
-
+  
+elseif("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "CYGWIN")
+  
+  # set(CMAKE_EXECUTABLE_FORMAT "ELF")
   set(SWIFT_HOST_VARIANT "windows" CACHE STRING
       "Deployment OS for Swift host tools (the compiler) [windows].")
+  configure_sdk_unix(CYGWIN "Cygwin" "windows" "cygwin" "${SWIFT_HOST_VARIANT_ARCH}" "${SWIFT_HOST_VARIANT_ARCH}-unknown-windows-cygnus")
```

Comment:
I'd also recommend substituting concrete values for `${SWIFT_HOST_VARIANT_ARCH}` back.